app/src/main/java/org/vss/impl/postgres/PostgresBackendImpl.java

```diff
@@ -66,13 +68,15 @@ public PutObjectResponse put(PutObjectRequest request) {
     List<VssDbRecord> vssRecords = new ArrayList<>(request.getTransactionItemsList().stream()
         .map(kv -> buildVssRecord(storeId, kv)).toList());
 
-    VssDbRecord globalVersionRecord = buildVssRecord(storeId,
-        KeyValue.newBuilder()
-            .setKey(GLOBAL_VERSION_KEY)
-            .setVersion(request.getGlobalVersion())
-            .build());
+    if (request.hasGlobalVersion()) {
```

Comment:
this doesn't increment the global version when it's not specified in the request.  we always want to increment it, right?

Reply:
I guess we could have the specifications say that it doesn't get incremented if not present in the request, but we should be explicit about it, since the developer might not expect that

Reply:
Based on how optimistic locking works, version needs to be provided by client in order to enact it.
So expectation is that we don't increment it and dont perform the check if developer doesn't supply it. 

That is it is a non-global-version-check-required write. 
We shouldn't be incrementing global_version in this case as client-side will not increment and has no way of knowing without performing a sync.

Based on different application needs, some might not need a global_version check on every write.